### PR TITLE
Reimplement transforms with OpenCV

### PR DESCRIPTION
**Pull request by QiJune**
I run the ImageTgzLoaderHeavy test in mac:

Before:

```bash
go test github.com/wangkuiyi/gotorch/vision/datasets -v -run ImageTgzLoaderHeavy -count=1

=== RUN   TestImageTgzLoaderHeavy
2020/09/15 17:28:09 throughput: 93.026710 samples/secs
--- PASS: TestImageTgzLoaderHeavy (4.64s)
PASS
ok  	github.com/wangkuiyi/gotorch/vision/datasets	4.783s
```

After using OpenCV:

```bash
=== RUN   TestImageTgzLoaderHeavy
2020/09/15 17:28:56 throughput: 177.355337 samples/secs
--- PASS: TestImageTgzLoaderHeavy (2.42s)
PASS
ok  	github.com/wangkuiyi/gotorch/vision/datasets	2.866s
```

OpenCV achieves nearly 2 times speedup.

### CODE REVIEW

**Review by wangkuiyi — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: cxwangyi/gotorch
+      - image: o0o0o/gotorch
```
Comment (QiJune) [inline .circleci/config.yml:6]: I push the CI image to my personal repo. Maybe we need a docker organization for GoTorch.

File: .travis.yml [hunk 2]
```diff
@@ -7,6 +7,13 @@ go:
   - 1.14
 os: osx
 
+addons:
+  homebrew:
+    packages:
+      - opencv
+      - pkgconfig
+    update: true
+
```
Comment (QiJune) [inline .travis.yml:16]: Installing OpenCV in macOS CI is time-consuming. It takes about 25 minutes.

Comment (wangkuiyi) [reply .travis.yml:16]: I guess it's the brew install command who takes a long time to run, right?

If so, how about we build OpenCV for macOS manually from source code (https://docs.opencv.org/master/d0/db2/tutorial_macos_install.html), and upload it to AWS or a GitHub repo, so we could install it from our source without using Homebrew?

File: .circleci/config.yml (hunk 1 above)
Comment (wangkuiyi) [reply .circleci/config.yml:6]: Good point. I created the DockerHub account gotorch.

File: .travis.yml (hunk 2 above)
Comment (QiJune) [reply .travis.yml:16]: I analyze the CI log, it takes about 5 min to update the Homebrew, and 15 min to install OpenCV.

OpenCV has a lots of dependencies, including: cmake, gflags, glog, metis, gmp, isl, mpfr, libmpc, gcc, openblas, gdbm, openssl@1.1, readline, sqlite, xz, python@3.8, tbb, suite-sparse, ceres-solver, aom, dav1d, freetype, frei0r, bdw-gc, libffi, libtool, pkg-config, guile, gettext, libtasn1, nettle, p11-kit, libevent, unbound, gnutls, lame, fribidi, pcre, glib, pixman, cairo, gobject-introspection, graphite2, icu4c, harfbuzz, libass, libbluray, libsoxr, libvidstab, libogg, libvorbis, libvpx, opencore-amr, jpeg, little-cms2, opus, rav1e, rtmpdump, flac, libsndfile, libsamplerate, rubberband, sdl2, snappy, speex, srt, webp, leptonica, tesseract, theora, x264, x265, xvid, ffmpeg, numpy, ilmbase, openexr, protobuf, boost, hdf5, netcdf, qt, sip, pyqt and vtk

I am afraid that we have to install lots of dependencies before building from source code. Let me have a try.

**Review by wangkuiyi — APPROVED** [review]

**Pull request merged**